content/200-concepts/100-components/02-prisma-client/053-client-extensions/150-type-utilities.mdx

```diff
@@ -23,50 +23,81 @@ The type utilities available are:
 - `Result<Type, Arguments, Operation>`: Takes the input arguments and provides the result for a given model and operation. You would usually use this in conjunction with `Args`. As with `Args`, `Result` helps you to re-use existing types to extend or modify them.
 - `Payload<Type, Operation>`: Retrieves the entire structure of the result, as scalars and relations objects for a given model and operation. For example, you can use this to determine which keys are scalars or objects at a type level.
 
-## Example
-
-The following example creates a new operation, `customCall`, based on `findFirst`. It has all of the arguments that `findFirst` has, with an additional property, `customProperty`.
+The following example creates a new operation, `exists`, based on `findFirst`. It has all of the arguments that `findFirst`.
 
 ```ts
-const prisma = new PrismaClient()
+const prisma = new PrismaClient().$extends({
+  model: {
+    $allModels: {
+      // Define a new `exists` operation on all models
+      // T is a generic type that corresponds to the current model
+      async exists<T>(
+        // `this` refers to the current type, e.g. `prisma.user` at runtime
+        this: T,
+
+        // The `exists` function will use the `where` arguments from the current model, `T`, and the `findFirst` operation
+        where: Prisma.Args<T, 'findFirst'>['where']
+      ): Promise<boolean> {
+        // Retrieve the current model at runtime
+        const context = Prisma.getExtensionContext(this)
 
-const xprisma = prisma.$extends({
+        // Prisma Client query that retrieves data based
```

Comment:
Should we place these same comments in the other code example?

Reply:
I've removed the duplicate comments from the second example. :)